sdk/python/sawtooth_sdk/workload/workload_generator.py

```diff
@@ -193,30 +194,43 @@ def on_new_batch(self, batch_id, stream):
             with self._lock:
                 self._submitted_batches_sample += 1
                 self._pending_batches.append(
-                    PendingBatch(id=batch_id, stream=stream))
+                    PendingBatch(id=batch_id, url=url))
+
+    def _status_request(self, batch_id, url):
+        data = json.dumps(batch_id).encode()
+        headers = {'Content-Type': 'application/json'}
+        headers['Content-Length'] = '%d' % len(data)
+
+        request = urllib.Request(
+            url + '/batch_status',
+            data,
+            headers=headers,
+            method='POST')
 
-    def _get_batch_status(self, batch_id, stream):
-        """
-        Sends a ClientBatchStatusRequest to the stream that the batch was
-        submitted to
-        """
-        request = ClientBatchStatusRequest(batch_ids=[batch_id])
-        future = stream.send(
-            message_type=Message.CLIENT_BATCH_STATUS_REQUEST,
-            content=request.SerializeToString())
         try:
-            result = future.result(timeout=0.5)
-            response = ClientBatchStatusResponse()
-            response.ParseFromString(result.content)
-            return ClientBatchStatusResponse.BatchStatus.Name(
-                response.batch_statuses[batch_id])
+            result = urllib.urlopen(request)
+            code, json_result = \
+                result.status, json.loads(result.read().decode())
 
-        except ValidatorConnectionError:
+            if code == 200 or code == 201 or code == 202:
+                status = list(json_result["data"].values())[0]
+                return status
+            else:
+                LOGGER.debug("(%s): %s", code, json_result)
+                return "UNKNOWN"
+        except HTTPError as e:
+            if "The connection to the validator was lost" in str(e.msg):
```

Comment:
My error PR will simplify the error checking you did here (and break it). Since your PR is basically ready to merge, let's make sure it gets in first, and then I'll add the necessary refactor to mine.